### ISSUE TITLE
Permissions error when calling end points [User: anonymous is not authorized to perform]

### ISSUE THREAD

**Issue by ryantomaselli** [samvera-labs/serverless-iiif#58]
Hey guys;

I'm getting a perms issue that results in a 403 when trying to make calls to any of the image server end points.

It seems like in a previous iteration of serverless-iiif I didn't have to add any special configuration to make calls to the image server just work.

If I CURL to the end point I get the following:
"Message":"User: anonymous is not authorized to perform: execute-api:Invoke on resource: arn:aws:execute-api:us-east-1:********4595:gdaw62mwig/latest/GET/iiif/2/repositories%2Fryanlocaldec222%2F2022-03%2FPfgbA76SdmegALIrJ3SA_jimmy.jpg/info.json"

If anyone can speak to why I might be getting this with the latest code I would appreciate it.

Thanks!

**Comment by mbklein** [comment]
Are you trying to invoke the API Gateway directly, or through CloudFront? What does your API Gateway Resource Policy look like? (It's visible from the AWS API Gateway console when the IIIF API is selected)

I just looked and noticed that the deploy script is setting the resource policy as if the API is being called through CloudFront. I'll fix that as soon as I have a chance, but in the meantime, you can manually edit your API's Resource Policy to:

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": "*",
            "Action": "execute-api:Invoke",
            "Resource": "arn:aws:execute-api:us-east-1:********4595:gdaw62mwig/*/*/*"
        }
    ]
} 
```
(Obviously you'll need to change your AWS account ID to the right one.)

**Comment by ryantomaselli** [comment]
Thanks @mbklein!

That change did allow the `/iiif/2/{id}/info.json` to respond successfully but now I get a 403 on the `/iiif/2/{id}/{proxy+}` end point.

Any ideas what may cause that?

**Comment by ryantomaselli** [comment]
@mbklein I found my issue...which oddly didn't exist in code gone by. It has to do with the manifest call `/iiif/2/{id}/info.json` not including the API stage in the manifest `@id` that is the URL for the get image call `/iiif/2/{id}/{proxy+}`.

Manifest URL:
`https://xxxxxx.execute-api.us-east-1.amazonaws.com/latest/iiif/2/repositories%2Fryanlocaldec222%2F2022-`03%2FPfgbA76SdmegALIrJ3SA_jimmy.jpg/info.json

Image URL:
`https://xxxxxx.execute-api.us-east-1.amazonaws.com/iiif/2/repositories%2Fryanlocaldec222%2F2022-03%2FPfgbA76SdmegALIrJ3SA_jimmy.jpg/full/378,/0/default.jpg
`

**Comment by ryantomaselli** [comment]
I just wanted to document the changes I made to the default application after the create function has run (in order to get things to work)...I accepted all the defaults for the create function parameters (all I entered was the Application name and SourceBucket)

Change 1:
Modify index.js in the IiifFunction lambda to address CORS issues

```
const makeResponse = (result) => {
  const { isBase64 } = helpers;

  const base64 = isBase64(result);
  const content = base64 ? result.body.toString('base64') : result.body;

  return {
    statusCode: 200,
    headers: {
      'Content-Type': result.contentType,
      'Access-Control-Allow-Origin': 'https://yourdomain.com',
      'Access-Control-Allow-Credentials': 'true'
    },
    isBase64Encoded: base64,
    body: content
  };
};
```

Modify resolver.js in the IiifFunction lambda to to not assume tif files
```
const defaultStreamLocation = (id) => {
  const key = id;
  return { Bucket: sourceBucket, Key: key };
};
```
Modify environment variable in the IiifFunction lambda so that manifest image path includes the stage
Update the `includeStage` environment variable to true

Update the resource policy on the IiifApi API gateway to not assume userAgent is cloudfront
Remove the following:
"Condition": {
    "StringEquals": {
        "aws:UserAgent": "Amazon CloudFront"
    }
}

**Comment by mbklein** [comment]
Thanks for diving into all of that. The `includeStage` bit and how it affects CloudFront forwarding and the back end Lambda has been an ongoing issue ever since I started trying to support both the bare API Gateway and CloudFront deployment modes in a single package. I’m starting to think that maybe they need to be separate SAM applications (deployed from the same repo) backed by the same Lambda code. The SAM template is just getting too complex to keep track of all the things that have to be set differently between the two modes.

**Comment by mbklein** [comment]
@ryantomaselli I believe this issue will be fixed by the switch from API Gateway to a Lambda Function URL, which never includes a stage, and also the splitting of the SAM deploy template into separate standalone and cache-enabled versions. The new versions aren't available in SAR yet because of a lag in SAR supporting Function URL-enabled Lambdas, but it would be great to get your feedback on it with respect to this issue once it's available.

**Comment by mbklein** [comment]
I believe this issue has been resolved by #65. Closing as fixed. Feel free to reopen if you encounter any further issues.
